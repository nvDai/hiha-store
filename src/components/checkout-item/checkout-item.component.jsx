import React from 'react'
import { connect } from 'react-redux'

import {
	clearItemFromCart,
	addItem,
	removeItem,
} from '../../redux/cart/cart.actions'

import './checkout-item.styles.scss'

export const CheckoutItem = ({
	item: { id, name, imageUrl, price, quantity },
	clearItemFromCart,
	addItem,
	removeItem,
}) => (
	<div className='checkout-item'>
		<div className='image-container'>
			<img src={imageUrl} alt='cart item' />
		</div>
		<span className='name'>{name}</span>
		<div className='quantity'>
			<div
				id='decrease-btn'
				className='arrow'
				onClick={() => removeItem({ id })}
			>
				&#10094;
			</div>
			<span className='value'>{quantity}</span>
			<div id='increase-btn' className='arrow' onClick={() => addItem({ id })}>
				&#10095;
			</div>
		</div>
		<span className='price'>$ {price}</span>
		<div className='remove-button' onClick={() => clearItemFromCart({ id })}>
			&#10005;
		</div>
	</div>
)

const mapDispatchToProps = dispatch => ({
	clearItemFromCart: item => dispatch(clearItemFromCart(item)),
	addItem: item => dispatch(addItem(item)),
	removeItem: item => dispatch(removeItem(item)),
})

export default connect(null, mapDispatchToProps)(CheckoutItem)
