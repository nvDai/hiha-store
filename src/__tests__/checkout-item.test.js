import React from 'react'

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { CheckoutItem } from '../components/checkout-item/checkout-item.component'

describe('CheckoutItem', () => {
	let wrapper, props
	beforeEach(() => {
		props = {
			item: {
				id: '1',
				name: 'T shirt',
				imageUrl: 'image.jpeg',
				price: 0,
				quantity: 0,
			},
			clearItemFromCart: jest.fn(),
			addItem: jest.fn(),
			removeItem: jest.fn(),
		}
		wrapper = shallow(<CheckoutItem {...props} />)
	})

	it('should render without crashing', () => {
		expect(toJson(wrapper)).toMatchSnapshot()
	})

	it('should increase quantity of item when clicked', () => {
		wrapper.find('#increase-btn').simulate('click')
		expect(props.addItem).toHaveBeenCalledTimes(1)
	})

	it('should decrease quantity of item when clicked', () => {
		wrapper.find('#decrease-btn').simulate('click')
		expect(props.removeItem).toHaveBeenCalledTimes(1)
	})

	it('should remove a item when clicked', () => {
		wrapper.find('.remove-button').simulate('click')
		expect(props.clearItemFromCart).toHaveBeenCalledTimes(1)
	})
})
