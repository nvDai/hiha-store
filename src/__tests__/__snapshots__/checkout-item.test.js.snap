// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutItem should render without crashing 1`] = `
<div
  className="checkout-item"
>
  <div
    className="image-container"
  >
    <img
      alt="cart item"
      src="image.jpeg"
    />
  </div>
  <span
    className="name"
  >
    T shirt
  </span>
  <div
    className="quantity"
  >
    <div
      className="arrow"
      id="decrease-btn"
      onClick={[Function]}
    >
      ❮
    </div>
    <span
      className="value"
    >
      0
    </span>
    <div
      className="arrow"
      id="increase-btn"
      onClick={[Function]}
    >
      ❯
    </div>
  </div>
  <span
    className="price"
  >
    $ 
    0
  </span>
  <div
    className="remove-button"
    onClick={[Function]}
  >
    ✕
  </div>
</div>
`;
